apis/task.py

```diff
@@ -271,11 +268,107 @@ class GpuTask(BaseTask):
   """This is a class to set up tasks for GPU.
 
   Attributes:
-    image: the image version that a GPU runs.
     image_project: the project that an image belongs to.
     image_family: the family group that an image belongs to.
   """
 
-  image: str
   image_project: str
   image_family: str
+  task_test_config: test_config.TestConfig[test_config.Gpu]
+  task_gcp_config: gcp_config.GCPConfig
+  task_metric_config: Optional[metric_config.MetricConfig] = None
+
+  def run(self) -> DAGNode:
+    """Run a test job.
+
+    Returns:
+      A task group with the following tasks chained: provision, run_model,
+      post_process and clean_up.
+    """
+    # piz: We skip the queued resource for GPU for now since there is no queued resource command for GPU.
+    with TaskGroup(
+        group_id=self.task_test_config.benchmark_id, prefix_group_id=True
+    ) as group:
+      provision, queued_resource, ssh_keys = self.provision()
```

Comment:
If tehre is no Queued Resource for GPU, we could update those `queued_resource` to something like `resource`. Please check other variable names in this file.

Reply:
Sounds good, updated